MonMaraicher/Features/Map/FarmerService.swift

```diff
@@ -7,21 +7,27 @@
 
 import Foundation
 
+enum LoadFarmerError: Error {
+    case invalidJsonFile
+    case errorDecoding
+}
+
 final class FarmerService {
 
-    func loadFarmers() -> [Farmer] {
+    func loadFarmers(handlerError: (Result<[Farmer], LoadFarmerError>) -> Void) {
```

Comment:
Suggestion: 
You use Result which totally fine! Nicely done btw! But here inside your func you are using a throwing func by using `try` and `catch`. 
Exercise: make that func throwing and not use Result. You will see the benefits ;)

Reply:
The name of the parameter `handleError` is not good. Your are not handling the error only right? It is used also when no error and it gives the data [Farmer]!

Another naming could be `resultHandler` (because your type if Result ;)). Or `completionHandler`...